src/main/java/com/myFirstProject/myFirstProject/service/PromoCodeServiceImpl.java

```diff
@@ -0,0 +1,54 @@
+package com.myFirstProject.myFirstProject.service;
+
+import com.myFirstProject.myFirstProject.converter.ReqConverterService;
+import com.myFirstProject.myFirstProject.dto.PromoCodeReq;
+import com.myFirstProject.myFirstProject.model.PromoCode;
+import com.myFirstProject.myFirstProject.repository.PromoCodeRepository;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+
+@Service
+public class PromoCodeServiceImpl implements PromoCodeService {
+    private Logger logger = LoggerFactory.getLogger(PaymentServiceImpl.class);
+    private PromoCodeRepository promoCodeRepository;
+    private ReqConverterService reqConverterService;
+
+    @Autowired
+    public void setPromoCodeRepository(PromoCodeRepository promoCodeRepository) {
+        this.promoCodeRepository = promoCodeRepository;
+    }
+
+    @Autowired
+    public void setReqConverterService(ReqConverterService reqConverterService) {
+        this.reqConverterService = reqConverterService;
+    }
+
+    @Transactional
+    @Override
+    public void deleteExpiredPromoCode(LocalDateTime time) {
+        List<PromoCode> promoCodes = promoCodeRepository.findByExpiredDateLessThan(time);
+        logger.info(String.format("Find %d promo codes with expired date", promoCodes.size()));
+        List<PromoCode> promoCodeForDelete = new ArrayList<>();
+        promoCodes.stream().filter(promoCode -> promoCode.getValid().equals(true))
```

Comment:
1) wrong stream expression:
 List<PromoCode> promoCodeForDelete = promoCodes.stream()
.filter(promoCode -> promoCode.getValid().equals(true))
.collect(toList);

also please follow my code style for lambda

2) promoCode.getValid().equals(true) - promoCode.getValid() can return true or false without additional checking. 
it should be:
filter(PromoCode::getValid)

Reply:
done